osmosis-installer.py

```diff
@@ -5,17 +5,64 @@
+
+osmo_home = subprocess.run(["echo $HOME/.osmosisd"], capture_output=True, shell=True, text=True).stdout.strip()
+
+parser = argparse.ArgumentParser(description="Osmosis Installer",formatter_class=fmt)
+auto = parser.add_argument_group('Automated')
+auto.add_argument('-m', '--mainnet-default', action='store_true', help='R|Use all default settings with no input for mainnet\n ', dest="mainnetDefault")
+auto.add_argument('-t', '--testnet-default', action='store_true', help='R|Use all default settings with no input for testnet\n ', dest="testnetDefault")
+
+both = parser.add_argument_group('Mainnet and Testnet')
+# mainnet and testnet
+both.add_argument('-s', '--swap', type = bool, default=True, help='R|Use swap if less than 32Gb RAM are detected \nDefault (bool): True\n ', dest="swapOn")
+both.add_argument('-i', '--install-home', type = str, default=osmo_home, help='R|Osmosis installation location \nDefault: "'+osmo_home+'"\n ', dest="installHome")
+both.add_argument('-na', '--name', type = str, default="defaultNode", help='R|Node name \nDefault: "defaultNode"\n ', dest="nodeName")
+args = 'tcp://0.0.0.0:1317;0.0.0.0:9090;0.0.0.0:9091;tcp://127.0.0.1:26658;tcp://127.0.0.1:26657;tcp://0.0.0.0:26656;localhost:6060'; both.add_argument('-p', '--ports', type=lambda s: [str(item) for item in s.split(';')], default=args, help='R|Single string seperated by semicolons of ports. Order must be api, grpc server, grpc web, abci app addr, rpc laddr, p2p laddr, and pprof laddr \nDefault: \"'+args+'\"\n ', dest="ports")
+args = ['full', 'client']; both.add_argument('-ty', '--type', type = str, choices=args, default='full', help='R|Node type \nDefault: "full" '+str(args)+'\n ', dest="nodeType")
+args = ['osmosis-1', 'osmo-test-4']; both.add_argument('-n', '--network', type = str, choices=args, default='osmosis-1', help='R|Network to join \nDefault: "osmosis-1" '+str(args)+'\n ', dest="network")
+args = ['default', 'nothing', 'everything']; both.add_argument('-pr', '--prune', type = str, choices=args, default='everything', help='R|Pruning settings \nDefault: "everything" '+str(args)+'\n ', dest="pruning")
+args = ['cosmoservice', 'osmoservice', 'noservice']; both.add_argument('-cvs', '--cosmovisor-service', type = str, choices=args, default='osmoservice', help='R|Start with cosmovisor systemctl service, osmosisd systemctl service, or exit without creating or starting a service \nDefault: "osmoservice" '+str(args), dest="cosmovisorService")
+testnet = parser.add_argument_group('Testnet only')
+#testnet
+args = ['snapshot', 'exit']; testnet.add_argument('-dst', '--data-sync-test', type = str, choices=args, default='snapshot', help='R|Data sync options \nDefault: "snapshot" '+str(args)+'\n ', dest="dataSyncTestnet")
+args = ['pruned', 'archive']; testnet.add_argument('-stt', '--snapshot-type-test', type = str, choices=args, default='pruned', help='R|Snapshot type \nDefault: "pruned" '+str(args)+'\n ', dest="snapshotTypeTestnet")

```

Comment:
I'm not super familiar with Python formatting rules but I'm wondering if it is possible to format these somehow? E.g. split into separate lines for readability?

Reply:
Do you mean specifically separate the args from the add_argument lines?

Reply:
Yes, and change the names, for example:
```
args = ['default', 'nothing', 'everything']; both.add_argument('-pr', '--prune', type = str, choices=args, default='everything', help='R|Pruning settings \nDefault: "everything" '+str(args)+'\n ', dest="pruning")
```

```
pruningStrateegyArgChoices = [
     'default', 
    'nothing', 
    'everything'
]
both.add_argument('-pr', '--prune', type = str, choices=pruningStrateegyArgChoices, default='everything', help='R|Pruning settings \nDefault: "everything" '+str(args)+'\n ', dest="pruning")
```

Reply:
An idiomatic way to improve readability in python could be:

```python
both.add_argument(
    '-s', 
    '--swap', 
    type = bool, 
    default=True, 
    help='R|Use swap if less than 32Gb RAM are detected \nDefault (bool): True\n ', 
    dest="swapOn")
```